src_c/key.c

```diff
@@ -68,6 +68,7 @@ key_get_repeat(PyObject *self, PyObject *_null)
 #define _PG_SCANCODEWRAPPER_TYPE_NAME "ScancodeWrapper"
 #define _PG_SCANCODEWRAPPER_TYPE_FULLNAME \
     "pygame.key." _PG_SCANCODEWRAPPER_TYPE_NAME
+#define NUM_KEYS 512
```

Comment:
Getting this wrong is extremely risky. Imagine an SDL scancode comes in with an id of 550, it tries to put that in the array and the entirety of pygame segfaults.

It looks like you can get this a constant from SDL in case they ever change it: https://github.com/libsdl-org/SDL/blob/main/include/SDL_scancode.h#L427 (SDL_NUM_SCANCODES)